### PR TITLE
[Terms of service] Sign user out and show toast when ToS cannot be loaded

### LINKED ISSUE #2657: [Bug] Terms of service screen is skipped if network is not available when the app is opened for the first time.
**Describe the bug**
Terms of service screen is skipped if network is not available. Instead it skips to the survey selection screen and the terms of service screen is showed on the next app reopen.

### PR DESCRIPTION
**Pull request by gino-m**
To simplify error handling, this PR also stops loading ToS from remote before directing to the fragment; instead, we always direct to the fragment. An empty ToS will be shown if not found.

Fixes #2657 

@shobhitagarwal1612 @sufyanAbbasi @scolsen or @anandwana001 PTAL?

### CODE REVIEW

**Comment by gino-m** [comment]
Looks like `MainViewModelTest` and `TermsOfServiceRepositoryTest` need to be updated.

File: ground/src/main/java/com/google/android/ground/MainActivity.kt [hunk 1]
```diff
@@ -58,7 +60,9 @@ class MainActivity : AbstractActivity() {
   @Inject lateinit var activityStreams: ActivityStreams
   @Inject lateinit var viewModelFactory: ViewModelFactory
   @Inject lateinit var navigator: Navigator
+  @Inject lateinit var authManager: AuthenticationManager
   @Inject lateinit var userRepository: UserRepository
+  @Inject lateinit var popups: EphemeralPopups
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/MainActivity.kt:65]: Can we remove these unused dependencies?

**Review by shobhitagarwal1612 — CHANGES_REQUESTED** [review]

File: ground/src/main/java/com/google/android/ground/MainActivity.kt (hunk 1 above)
Comment (gino-m) [reply ground/src/main/java/com/google/android/ground/MainActivity.kt:65]: Done

**Review by shobhitagarwal1612 — APPROVED** [review]

**Pull request merged**